sidecar/src/figwheel_sidecar/repl.clj

```diff
@@ -251,23 +253,33 @@
        #_(when (in-nrepl-env?)
            (throw (ex-info "Hey" {})))))))
 
-(defn repl
+(defn cljs-repl-env
   ([build figwheel-server]
-   (repl build figwheel-server {}))
+   (cljs-repl-env build figwheel-server {}))
   ([build figwheel-server opts]
    (let [opts (merge (assoc (or (:compiler build) (:build-options build))
                             :warn-on-undeclared true
-                            :eval #'catch-warnings-and-exceptions-eval-cljs
-                            )
+                            :eval #'catch-warnings-and-exceptions-eval-cljs)
                      opts)
          figwheel-server (assoc figwheel-server
                                 :repl-print-chan (chan))
          figwheel-repl-env (repl-env figwheel-server build)
-         repl-opts (assoc opts :compiler-env (:compiler-env build))
-         protocol (if (in-nrepl-env?)
-                    :nrepl
-                    :default)]
-     (start-cljs-repl protocol figwheel-repl-env repl-opts))))
+         repl-opts (merge
+                     {:compiler-env (:compiler-env build)
+                      :special-fns cljs.repl/default-special-fns
+                      :output-dir "out"}
+                     opts)]
+     (assoc figwheel-repl-env
+            ;; these are merged to opts by cljs.repl/repl*
+            :repl-opts repl-opts))))
```

Comment:
This was added in the same commit as the IReplEnvOptions. `cljs.repl/repl*` doesn't directly use this `:repl-opts` key but instead calls the `-repl-options` method of the protocol and after this PR `FigwheelRepl` record implements that method and uses `repl-opts` property.